### DISCUSSION TITLE
Contest Logging additions

### DISCUSSION THREAD

**Discussion by M0TZO** [magicbug/Cloudlog#858] [Ideas / Feature Requests]
Was on an online club chat tonight, and I was showing them the Contest Logging feature. Was a couple of thoughts that was mentioned that would be nice, if possible.

- Dupe Checking
- The Contest Log at bottom to return (_somehow_) if leave the page

**Comment by magicbug** [comment]
Errm last item is already implemented as long as you select a contest name

Dupes should be able to add.

**Comment by on4dgt** [comment]
Can I add the possibility to use serial+ other.
In a UBA contest I always struggle with things like _" ON stations: RS(T) + serial number starting from 001 + Belgian province. The province is part of the exchange and must be mentioned in the log. "_
Which mostly results in me handing out dupe serials .

**Reply by magicbug** [reply]
Perfectly ok request, I'm also wanting serial + gridsquare for VHF contests.

**Reply by smurphboy** [reply]
I second this. I was doing an RSGB contest on Tuesday and would really like to be able to customise the contest input form by contest (i.e. select what the exchange is, i.e. RS(T) + Serial + Gridsquare.

**Comment by myzinsky** [comment]
The current contest list does not seam to be the full international list. 

Why not using this one from CloudLogOffline: 

```
                    "10-10-FALL-CW",
                    "10-10-FALL-DIGITAL",
                    "10-10-OPEN-SEASON",
                    "10-10-SPIRIT-OF-76",
                    "10-10-SPRING-CW",
                    "10-10-SPRING-DIGITAL",
                    "10-10-SPRINT",
                    "10-10-SUMMER-PHONE",
                    "10-10-WINTER-PHONE",
                    "7QP",
                    "9A-CW",
                    "AADX-CW",
                    "AADX-SSB",
                    "AGCW-QRP",
                    "AL-QSO-PARTY",
                    "ALL-AFRICA",
                    "AP-SPRINT",
                    "AR-QSO-PARTY",
                    "ARI-DX",
                    "ARRL-10",
                    "ARRL-10-GHZ",
                    "ARRL-160",
                    "ARRL-222",
                    "ARRL-DX-CW",
                    "ARRL-DX-SSB",
                    "ARRL-EME",
                    "ARRL-FD",
                    "ARRL-HPM-150",
                    "ARRL-RR-CW",
                    "ARRL-RR-DIG",
                    "ARRL-RR-PH",
                    "ARRL-RTTY",
                    "ARRL-SCR",
                    "ARRL-SS-CW",
                    "ARRL-SS-SSB",
                    "ARRL-VHF-JAN",
                    "ARRL-VHF-JUN",
                    "ARRL-VHF-SEP",
                    "AVHFC",
                    "AZ-QSO-PARTY",
                    "BALKAN-HF",
                    "BALTIC-CONTEST",
                    "BARTG-RTTY",
                    "BARTG-SPRINT",
                    "BATAVIA",
                    "BC-QSO-PARTY",
                    "BUCURESTI-DIGITAL",
                    "CA-QSO-PARTY",
                    "CANADA-DAY",
                    "CANADA-WINTER",
                    "COQP",
                    "CQ-160-CW",
                    "CQ-160-SSB",
                    "CQ-M",
                    "CQ-VHF",
                    "CQ-WPX-CW",
                    "CQ-WPX-RTTY",
                    "CQ-WPX-SSB",
                    "CQ-WW-CW",
                    "CQ-WW-RTTY",
                    "CQ-WW-SSB",
                    "CQMMDX",
                    "CVA-DX-CW",
                    "CVA-DX-SSB",
                    "CW-OPEN",
                    "CW-OPS",
                    "DARC-10",
                    "DARC-WAEDC-CW",
                    "DARC-WAEDC-RTTY",
                    "DARC-WAEDC-SSB",
                    "DE-QSO-PARTY",
                    "DL-DX-RTTY",
                    "DRCG-WW-RTTY",
                    "EA-MAJESTAD-CW",
                    "EA-MAJESTAD-SSB",
                    "EA-PSK",
                    "EARTTY",
                    "EU-PSK-DX",
                    "EUCW-160",
                    "EUHFC",
                    "EURASIA-CHAMP",
                    "F9AA-CW",
                    "F9AA-DIGI",
                    "F9AA-SSB",
                    "FCG-FQP",
                    "FT8-RU",
                    "GA-QSO-PARTY",
                    "HA-DX",
                    "HELVETIA",
                    "HI-QSO-PARTY",
                    "HOLYLAND",
                    "IAQP",
                    "IARU-HF",
                    "ID-QSO-PARTY",
                    "IL-QSO-PARTY",
                    "IN-QSO-PARTY",
                    "ISLAND-QSO-PARTY",
                    "JARTS-WW-RTTY",
                    "JIDX-CW",
                    "JIDX-SSB",
                    "KANHAM",
                    "KCJ-TOPBAND",
                    "KS-QSO-PARTY",
                    "KYQP",
                    "LA-QSO-PARTY",
                    "LZ-DX",
                    "MAKROTHEN-RTTY",
                    "MDC-QSO-PARTY",
                    "ME-QSO-PARTY",
                    "MI-QSO-PARTY",
                    "MMC-HF-CW",
                    "MN-QSO-PARTY",
                    "MO-QSO-PARTY",
                    "MRRC-OHQP",
                    "MS-QSO-PARTY",
                    "NA-SPRINT-CW",
                    "NA-SPRINT-RTTY",
                    "NA-SPRINT-SSB",
                    "NAQP-CW",
                    "NAQP-RTTY",
                    "NAQP-SSB",
                    "NC-QSO-PARTY",
                    "NCCC-SPRINT-CW",
                    "NCCC-SPRINT-RTTY",
                    "ND-QSO-PARTY",
                    "NE-QSO-PARTY",
                    "NEQP",
                    "NH-QSO-PARTY",
                    "NJQP",
                    "NM-QSO-PARTY",
                    "NRAU-10",
                    "NV-QSO-PARTY",
                    "NYQP",
                    "OCEANIA-DX-CW",
                    "OCEANIA-DX-SSB",
                    "OK-DX-RTTY",
                    "OK-OM-DX",
                    "OK-QSO-PARTY",
                    "ON-QSO-PARTY",
                    "PA-QSO-PARTY",
                    "PACC",
                    "POC",
                    "PORTUGAL-DAY",
                    "RADIO-160",
                    "RADIO-ONY",
                    "RADIO-WW-RTTY",
                    "RAEM",
                    "RCC-CUP",
                    "RDAC",
                    "RDXC",
                    "REF-CW",
                    "REF-SSB",
                    "RSGB-160",
                    "RSGB-80M-AUT",
                    "RSGB-80M-CC",
                    "RSGB-AFS-CW",
                    "RSGB-COMMONWEALTH",
                    "RSGB-DX",
                    "RSGB-FT4",
                    "RSGB-HQP",
                    "RSGB-IOTA",
                    "RSGB-LOW-POWER",
                    "RSGB-NFD",
                    "RSGB-ROLO",
                    "RTTYOPS-WEEKEND-SPRINT",
                    "RTTYOPS-WEEKSPRINT",
                    "RTTYOPS-WW-RTTY",
                    "RUS-WW-DIGI",
                    "RUS-WW-MM",
                    "RUS-WW-PSK",
                    "SA10M",
                    "SAC-CW",
                    "SAC-SSB",
                    "SARTG-NY-RTTY",
                    "SARTG-RTTY",
                    "SC-QSO-PARTY",
                    "SDQSOP",
                    "SPDX",
                    "SPDXC",
                    "SPDXC-RTTY",
                    "STAYHOME",
                    "STEW-PERRY",
                    "TARA-RTTY",
                    "TESLA-HF",
                    "TISZACUP",
                    "TN-QSO-PARTY",
                    "TRC-DX",
                    "TXQP",
                    "UBA-DX-CW",
                    "UBA-DX-SSB",
                    "UBA-ON-2M",
                    "UBA-ON-6M",
                    "UBA-ON-CW",
                    "UBA-ON-SSB",
                    "UBA-PSK63-PREFIX",
                    "UBA-SPRING-CONTEST",
                    "UKEI-DX",
                    "UKEICC-80M",
                    "UKR-CHAMP-RTTY",
                    "UKRAINIAN-DX",
                    "UN-DX",
                    "UR-DX-DIGI",
                    "UR-DX-RTTY",
                    "VA-QSO-PARTY",
                    "VKSHIRES",
                    "VOLTA-RTTY",
                    "VT-QSO-PARTY",
                    "WA-SALMON-RUN",
                    "WAG",
                    "WAPC-DX",
                    "WFD",
                    "WIQP",
                    "WVQP",
                    "WW-DIGI",
                    "WW-PMC",
                    "WWSA",
                    "WWSAC",
                    "XE-RTTY",
                    "XMAS",
                    "YARC-QSO-PARTY",
                    "YB-DX-CONTEST",
                    "YL-OM",
                    "YUDX",
```

**Reply by dg9vh** [reply]
Just add the missing contests into the existing list via PR 😊 the list of contests in cloudlog is from original ADIF spec

**Comment by AndreasK79** [comment]
As @dg9vh said, just add a PR with them in the list. I was thinking about maybe adding a table for all the contests, so that it is editable inside Cloudlog, but it's something I never did write when working with it.

Perhaps that would be a better way after all. What do you all say?

**Reply by dg9vh** [reply]
Would be a good idea having the contest-table like the modes editable and stored in the database... So we can have a customizable select-list for each installation... Some OM would never be active in some digital or CW or SSB-contests (because of no interest).

What is @magicbug thinking about this idea?

We could fill the table (like the modes) initially with a migration-file and then let the dog run!

73 de Kim

**Reply by AndreasK79** [reply]
@dg9vh Not much response, but I'll add it anyway. I've setup the table and filled it with the existing contests right now. The Contest Logging now fetches from the database. 

I need to make a migration file and a gui for add/delete/edit contests.

**Comment by dg9vh** [comment]
Maybe a chance to sort the contests by name and not by ID :-)

Am Mi., 10. Feb. 2021 um 12:26 Uhr schrieb Andreas Kristiansen <
notifications@github.com>:

> @dg9vh <https://github.com/dg9vh> Not much response, but I'll add it
> anyway. I've setup the table and filled it with the existing contests right
> now. The Contest Logging now fetches from the database.
>
> I need to make a migration file and a gui for add/delete/edit contests.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/magicbug/Cloudlog/discussions/858#discussioncomment-356198>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADKN5OUBHBR57TD7QVWSU3TS6JUOLANCNFSM4WZR2E7A>
> .
>


-- 
Amateurfunk ist das, was uns verbindet! Bleiben wir in Kontakt!
Ortsverband Q03, Neunkirchen

**Reply by AndreasK79** [reply]
@dg9vh Yep. That's taken care of in the query:
`SELECT name, adifname FROM contest WHERE active = 1 ORDER BY name ASC`

**Reply by dg9vh** [reply]
I mean also sorting them in the select-option-list :-)

**Reply by AndreasK79** [reply]
You mean setting the sortorder yourself?

**Reply by dg9vh** [reply]
no, I mean sorting actually is by id (value in select-option) what is suboptimal if you are searching for a contest, because you see the name in the text-area of the option...

So sorting by name would be a bit more UX.

**Reply by AndreasK79** [reply]
Ok, that is done by fetching them from the database, sorted by name. Then the dropdown looks like this:
![image](https://user-images.githubusercontent.com/6977712/107559574-93b4aa00-6bdc-11eb-99b8-8b5ad816e5fa.png)

**Reply by dg9vh** [reply]
wonderful :-)

**Reply by AndreasK79** [reply]
And the view for the list is in place:
![image](https://user-images.githubusercontent.com/6977712/107560150-4a188f00-6bdd-11eb-9855-e869e14e70c8.png)
Still need to write the stuff behind the buttons.

**Comment by dg9vh** [comment]
looks very good :-)

**Comment by mbridak** [comment]
Can we by chance add the STX_STRING and SRX_STRING fields to the edit QSO dialog if those fields were left blank?
They appear in the edit dialog if the info was initially entered, but if it was forgotten there seems to be no way to add them after the fact, short of deleting qso and re-entering.

**Reply by saintaardvark** [reply]
I'd also like this, though my use case is a little different: I typically log to a paper notebook during a contest, then put them in the computer afterward.

**Comment by magicbug** [comment]
I've added most of the features i saw to the roadmap see https://github.com/magicbug/Cloudlog/projects/8 I might have missed a couple but I think I got it all.

Apart from adding scoring (unlikely) are these the only missing components?

**Reply by electricEngineer** [reply]
I tried out Cloudlog only a few days ago and am pretty impressed of the overall performance and the one of a kind feature to use the webbrowser for logging. This simplifies the task of logging for club stations.

For contesting I would be happy if its also possible to have a field for the DARC DOK. Also scoring would be nice as you said. Maybe its possible to have for every contest a List of fields in the database which describes the desired contest Page layout and a fallback layout if nothing is specified.

**Reply by magicbug** [reply]
It would be nice to have scoring and if someone had the time/interest could be driven by the contest id but sadly not something that's on my list yet maybe further down the road.

**Comment by AndreasK79** [comment]
Exchangetypes... getting there:
![image](https://user-images.githubusercontent.com/6977712/129243450-075caad5-0b31-4e6c-93f7-e4c2aa56dbd3.png)

**Reply by smurphboy** [reply]
Yes! This!!!

**Reply by electricEngineer** [reply]
Looks Nice! Could you add a field for DOK, Serial + DOK and Serial + DOK + Gridsquare please? This would be great!

**Reply by AndreasK79** [reply]
I'll see what I can do. First I need to get everything working with these changes.

**Reply by Cadair** [reply]
Yes please! This would be amazing.

**Reply by AndreasK79** [reply]
@Cadair These changes exists in the codebase, so feel free to try it out. DOK has not been added. It's in my list.

**Reply by AndreasK79** [reply]
@electricEngineer I've looked into adding DOK. The problem here, is that we have no place to store your own DOK. Is there a reason you can't use exchange sent/received?

**Reply by electricEngineer** [reply]
The DOK sent could be fetched from the station and written in the sent field. For me it only would be great if the other stations DOK could be stored directly in the contact so I can easier check e.g. status of the "Deutschland Diplom" of the DARC. Maybe it would be a good way to have a checkbox or dropdown on the contest edit page where the value in the received Exchange field would be automatically copied to. The DOK field should already exist in the database...

**Reply by AndreasK79** [reply]
I see your point. The received DOK field has a field in the database, but we have no field for sent DOK.
The better solution, as you say, would be to copy the received exchange to the DOK field, and keep it in received exchange as well.
I'll take a look at how that approach can be implemented.

**Reply by AndreasK79** [reply]
@electricEngineer if you get the latest code, I've added an option to copy the received exchange to the DOK field in the database.

**Comment by Cadair** [comment]
One thing that would be useful is being able to export to ADIF between specific times not just on dates. I did two contests back to back and needed an export for each separately.

**Reply by AndreasK79** [reply]
@Cadair agree. I think it should be pretty easy to add. I'll add it to the TODO-list.

**Comment by smurphboy** [comment]
I think there is a 'bug' in the contest mode when it comes to Gridsquare(S) field, it appears to read the account level gridsquare and not the gridsquare of the current station location. Which I think is the wrong behaviour?

**Reply by AndreasK79** [reply]
@smurphboy I see the problem now. The logging is correct, and fetches from the active station location.
What is shown on the screen when logging, is incorrect. That one is fetched from the logged in user account.
I'll make a note of it so that it gets fixed. Thanks for reporting.

**Reply by smurphboy** [reply]
No worries, I only noticed as I've moved QTH and tried another contest. It's a great feature though and very usable. I'll keep testing it

**Reply by AndreasK79** [reply]
@smurphboy this is fixed now in the codebase,

**Comment by AndreasK79** [comment]
A little teaser for the Contest Logging:
![image](https://user-images.githubusercontent.com/6977712/138587944-afbeb2c5-0513-41f8-afbb-26e9333508cd.png)

**Reply by myzinsky** [reply]
Worked before in the same contest? Same band? Is it configurable?

**Reply by AndreasK79** [reply]
I'm not planning on making if configurable. It's worked before, as in the same contest, band, mode and callsign.

Right now, I'm testing it out and working out any issues.

**Comment by dg9vh** [comment]
Worked before should be in common with the contest rules... Mostly it's
same band and mode in one and the same contest...

Dr. Matthias Jung ***@***.***> schrieb am So., 24. Okt. 2021,
11:24:

**Reply by AndreasK79** [reply]
Worked before, as in the same contest, band, mode and callsign.

**Comment by dg9vh** [comment]
That's ok :-) There are only a few contests allowing other possibilities...
However it should be possible to log the QSO even if it's a dupe! There
could be reasons to work someone twice...

Am So., 24. Okt. 2021 um 11:49 Uhr schrieb Andreas Kristiansen <
***@***.***>:

> I'm not planning on making if configurable. It's worked before, as in the
> same contest, band, mode and callsign.
>
> Right now, I'm testing it out and working out any issues.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/magicbug/Cloudlog/discussions/858#discussioncomment-1527113>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADKN5OWMLZWBKVLW77E3VQDUIPJDJANCNFSM4WZR2E7A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Amateurfunk ist das, was uns verbindet! Bleiben wir in Kontakt!
Ortsverband Q03, Neunkirchen

**Reply by AndreasK79** [reply]
Always work dupes! This is not stopping you, just informing.

**Comment by dg9vh** [comment]
;-) then it's ok :-)

Am So., 24. Okt. 2021 um 12:28 Uhr schrieb Andreas Kristiansen <
***@***.***>:

> Always work dupes! This is not stopping you, just informing.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/magicbug/Cloudlog/discussions/858#discussioncomment-1527178>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADKN5OXJVC5UKX4BOWO4F4LUIPNV7ANCNFSM4WZR2E7A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Amateurfunk ist das, was uns verbindet! Bleiben wir in Kontakt!
Ortsverband Q03, Neunkirchen

**Comment by AndreasK79** [comment]
I've merged dupechecking into the master now.
